src/NHibernate.Test/NHSpecificTest/GH3516/Entity.cs

```diff
@@ -0,0 +1,14 @@
+﻿using System;
+
+namespace NHibernate.Test.NHSpecificTest.GH3516
+{
+	public class Entity
+	{
+		public virtual Guid Id { get; set; }
+		public virtual string Name { get; set; }
+
+		public const string NameWithSingleQuote = "'; drop table Entity; --";
+
+		public const string NameWithEscapedSingleQuote = @"\'; drop table Entity; --";
```

Comment:
Some databases support backslash escaping on top of doubling escaping. So if the fix was only doubling the quote without also escaping the backslash, there would still be an injection vulnerability, because the first quote would be escaped and not considered by the database.